File: frontend/mobile_react_native/src/screens/ado/InscriptionScreen.js
import React, { useState } from 'react';
import { View, Text, TextInput, StyleSheet, ScrollView, Image, ActivityIndicator, TouchableOpacity } from 'react-native';
import { colors, fonts } from '../../theme/colors';
import { HeroBand, SectionLabel, PrimaryButton, GhostButton } from '../../components/Shared';
import { inscriptionAdo } from '../../services/apiService';

export default function InscriptionScreen({ navigation }) {
  const [pseudo, setPseudo] = useState('');
  const [age, setAge] = useState('');
  const [motDePasse, setMotDePasse] = useState('');
  const [confirmation, setConfirmation] = useState('');
  const [chargement, setChargement] = useState(false);
  const [erreur, setErreur] = useState(null);
  const [voirMdp, setVoirMdp] = useState(false);

  const creerCompte = async () => {
    if (!pseudo.trim()) { setErreur('Choisis un pseudo.'); return; }
    if (pseudo.trim().length < 3) { setErreur('Le pseudo doit faire au moins 3 caractères.'); return; }
    if (motDePasse !== confirmation) { setErreur('Les mots de passe ne correspondent pas.'); return; }
    const ageInt = parseInt(age, 10);
    if (!ageInt || ageInt < 10 || ageInt > 99) { setErreur('Âge invalide (10-99).'); return; }
    setChargement(true);
    setErreur(null);
    try {
      await inscriptionAdo({ pseudo: pseudo.trim(), age: ageInt, motDePasse, consentementAnalyseIa: true });
      navigation.reset({ index: 0, routes: [{ name: 'Accueil' }] });
    } catch (e) {
      const msg = e.message || 'Erreur de connexion.';
      if (msg.toLowerCase().includes('déjà pris') || msg.toLowerCase().includes('already')) {
        setErreur('Ce pseudo est déjà pris. Choisis-en un autre.');
      } else {
        setErreur(msg);
      }
    } finally {
      setChargement(false);
    }
  };

  return (
    <ScrollView style={s.container} contentContainerStyle={{ padding: 20 }}>
      <HeroBand
        titre="Crée ton espace"
        sousTitre="Anonyme, juste pour toi"
        leading={<View style={s.logoChip}><Image source={require('../../../assets/logo_sangolo.png')} style={s.logoSmall} resizeMode="contain" /></View>}
      />
      <SectionLabel>Pseudo</SectionLabel>
      <TextInput style={s.input} placeholder="Choisis un pseudo (unique)" value={pseudo} onChangeText={setPseudo} autoCapitalize="none" />
      <SectionLabel>Âge</SectionLabel>
      <TextInput style={s.input} placeholder="16" keyboardType="number-pad" value={age} onChangeText={setAge} />
      <SectionLabel>Mot de passe</SectionLabel>
      <View style={s.passwordRow}>
        <TextInput style={[s.input, { flex: 1 }]} placeholder="8 caractères minimum" secureTextEntry={!voirMdp} value={motDePasse} onChangeText={setMotDePasse} />
        <TouchableOpacity style={s.eyeBtn} onPress={() => setVoirMdp(!voirMdp)}>
          <Text style={s.eyeText}>{voirMdp ? '🙈' : '👁️'}</Text>
        </TouchableOpacity>
      </View>
      <SectionLabel>Confirmation</SectionLabel>
      <View style={s.passwordRow}>
        <TextInput style={[s.input, { flex: 1 }]} placeholder="Confirme ton mot de passe" secureTextEntry={!voirMdp} value={confirmation} onChangeText={setConfirmation} />
        <TouchableOpacity style={s.eyeBtn} onPress={() => setVoirMdp(!voirMdp)}>
          <Text style={s.eyeText}>{voirMdp ? '🙈' : '👁️'}</Text>
        </TouchableOpacity>
      </View>
      {erreur ? <Text style={s.erreur}>{erreur}</Text> : null}
      <View style={{ marginTop: 18 }}>
        {chargement ? <ActivityIndicator color={colors.ink} size="large" /> : (
          <PrimaryButton label="Créer mon espace" onPress={creerCompte} />
        )}
      </View>
      <Text style={s.note}>Aucune donnée identifiante requise</Text>
      <View style={{ marginTop: 12 }}>
        <GhostButton label="J'ai déjà un compte — Se connecter" onPress={() => navigation.navigate('ConnexionAdo')} />
      </View>
    </ScrollView>
  );
}

const s = StyleSheet.create({
  container: { flex: 1, backgroundColor: colors.paper },
  logoChip: { width: 60, height: 44, backgroundColor: colors.paper, borderRadius: 10, alignItems: 'center', justifyContent: 'center', marginRight: 10 },
  logoSmall: { width: 44, height: 30 },
  input: { backgroundColor: colors.card, borderRadius: 14, paddingHorizontal: 16, paddingVertical: 14, fontFamily: fonts.body, fontSize: 14, color: colors.ink },
  passwordRow: { flexDirection: 'row', alignItems: 'center', gap: 8 },
  eyeBtn: { width: 44, height: 44, borderRadius: 14, backgroundColor: colors.card, alignItems: 'center', justifyContent: 'center' },
  eyeText: { fontSize: 18 },
  erreur: { color: colors.coral, fontSize: 13, marginTop: 10, fontFamily: fonts.body },
  note: { textAlign: 'center', color: colors.inkSoft, fontSize: 12, marginTop: 10, fontFamily: fonts.body },
});

File: frontend/mobile/lib/screens/ado/inscription_screen.dart
import 'package:flutter/material.dart';
import '../../theme/app_theme.dart';
import '../../services/api_service.dart';
import 'accueil_screen.dart';

class InscriptionScreen extends StatefulWidget {
  const InscriptionScreen({super.key});

  @override
  State<InscriptionScreen> createState() => _InscriptionScreenState();
}

class _InscriptionScreenState extends State<InscriptionScreen> {
  final _pseudoController = TextEditingController();
  final _ageController = TextEditingController();
  final _mdpController = TextEditingController();
  final _confirmationController = TextEditingController();
  final _api = ApiService();
  bool _chargement = false;
  String? _erreur;

  Future<void> _creerCompte() async {
    if (_mdpController.text != _confirmationController.text) {
      setState(() => _erreur = 'Les mots de passe ne correspondent pas.');
      return;
    }
    final age = int.tryParse(_ageController.text);
    if (age == null) {
      setState(() => _erreur = 'Âge invalide.');
      return;
    }

    setState(() { _chargement = true; _erreur = null; });
    try {
      await _api.inscriptionAdo(
        pseudo: _pseudoController.text.trim(),
        age: age,
        motDePasse: _mdpController.text,
        consentementAnalyseIa: true,
      );
      if (!mounted) return;
      Navigator.of(context).pushReplacement(
        MaterialPageRoute(builder: (_) => const AccueilScreen()),
      );
    } catch (e) {
      setState(() => _erreur = e.toString());
    } finally {
      if (mounted) setState(() => _chargement = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.paper,
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(18),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: [
              Container(
                padding: const EdgeInsets.symmetric(vertical: 22),
                decoration: BoxDecoration(
                  gradient: AppColors.heroGradient,
                  borderRadius: BorderRadius.circular(22),
                ),
                child: Column(
                  children: [
                    Container(
                      width: 88, height: 56,
                      decoration: BoxDecoration(
                        color: AppColors.paper,
                        borderRadius: BorderRadius.circular(12),
                      ),
                      child: Center(
                        child: Container(
                          width: 64, height: 64,
                          decoration: BoxDecoration(
                            color: AppColors.amber,
                            borderRadius: BorderRadius.circular(12),
                          ),
                          child: const Center(
                            child: Text('S', style: TextStyle(
                              color: Colors.white, fontSize: 30, fontWeight: FontWeight.w700,
                            )),
                          ),
                        ),
                      ),
                    ),
                    const SizedBox(height: 10),
                    Text('Crée ton espace',
                        style: Theme.of(context).textTheme.titleLarge
                            ?.copyWith(color: Colors.white)),
                    Text('Anonyme, juste pour toi',
                        style: Theme.of(context).textTheme.bodySmall
                            ?.copyWith(color: Colors.white70)),
                  ],
                ),
              ),
              const SizedBox(height: 20),
              _label('Pseudo'),
              TextField(controller: _pseudoController,
                  decoration: const InputDecoration(hintText: 'Choisis un pseudo')),
              const SizedBox(height: 12),
              _label('Âge'),
              TextField(controller: _ageController, keyboardType: TextInputType.number,
                  decoration: const InputDecoration(hintText: '16')),
              const SizedBox(height: 12),
              _label('Mot de passe'),
              TextField(controller: _mdpController, obscureText: true,
                  decoration: const InputDecoration(hintText: '••••••••')),
              const SizedBox(height: 12),
              _label('Confirmation'),
              TextField(controller: _confirmationController, obscureText: true,
                  decoration: const InputDecoration(hintText: '••••••••')),
              if (_erreur != null) ...[
                const SizedBox(height: 10),
                Text(_erreur!, style: const TextStyle(color: AppColors.coral, fontSize: 12)),
              ],
              const SizedBox(height: 18),
              ElevatedButton(
                onPressed: _chargement ? null : _creerCompte,
                child: _chargement
                    ? const SizedBox(width: 18, height: 18,
                        child: CircularProgressIndicator(strokeWidth: 2, color: Colors.white))
                    : const Text('Créer mon espace'),
              ),
              const SizedBox(height: 10),
              Text('Aucune donnée identifiante requise',
                  textAlign: TextAlign.center,
                  style: Theme.of(context).textTheme.bodySmall),
            ],
          ),
        ),
      ),
    );
  }

  Widget _label(String text) => Padding(
        padding: const EdgeInsets.only(bottom: 4),
        child: Text(text, style: Theme.of(context).textTheme.labelSmall),
      );
}
